scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -192,6 +192,8 @@ def _output_handler(self, proc, harness):
             timeout_extended = False
             timeout_time = time.time() + self.timeout
             while True:
+                if proc.poll():
```

Comment:
would this be better to check the pro in reader handler? In my testing, the native_posix/qemu somtime crash in the middle of testing.

Reply:
+1 to check in the handler.

also, I suspect that with the proposed change it is possible to miss `proc` output on its successful end by the moment of this check.

@bearzly, could you please add a debug log entry and provide log snippets of how this change resolves the addressed problem.